test-data/unit/check-parameter-specification.test

```diff
@@ -416,3 +415,614 @@ with f() as x:
     pass
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-full.pyi]
+
+[case testParamSpecLiterals]
+from typing_extensions import ParamSpec, TypeAlias
+from typing import Generic, TypeVar
+
+P = ParamSpec("P")
+T = TypeVar("T")
+
+class Z(Generic[P]): ...
+
+# literals can be applied
+n: Z[[int]]
+
+# TODO: type aliases too
+nt1 = Z[[int]]
+nt2: TypeAlias = Z[[int]]
+
+unt1: nt1
+unt2: nt2
+
+# literals actually keep types
+reveal_type(n)  # N: Revealed type is "__main__.Z[[builtins.int]]"
+reveal_type(unt1)  # N: Revealed type is "__main__.Z[[builtins.int]]"
+reveal_type(unt2)  # N: Revealed type is "__main__.Z[[builtins.int]]"
+
+# passing into a function keeps the type
+def fT(a: T) -> T: ...
+def fP(a: Z[P]) -> Z[P]: ...
+
+reveal_type(fT(n))  # N: Revealed type is "__main__.Z*[[builtins.int]]"
```

Comment:
```suggestion
reveal_type(fT(n))  # N: Revealed type is "__main__.Z[[builtins.int]]"
```
The output for `reveal_type` was changed in #12459. It no longer includes `*` for inferred types.